test/services/cert/rhsm_client_test.rb

```diff
@@ -5,36 +5,48 @@
 module Katello
   class RhsmClientTest < Minitest::Test
     CERT = '
-      -----BEGIN CERTIFICATE-----
```

Comment:
It's critical that your change works in a backward compatible way. Best way to ensure that is to have two certs with the different subject formats and verify each of them with your new technique!